src/dmd/vsoptions.d

```diff
@@ -635,18 +722,46 @@ const(char)* detectVSInstallDirViaCOM()
         return null;
     scope(exit) instances.Release();
 
+    static struct WrappedBString
+    {
+        BSTR ptr;
+        this(this) @disable;
+        ~this() { SysFreeString(ptr); }
+        bool opCast(T : bool)() const { return ptr !is null; }
+        size_t length() { return SysStringLen(ptr); }
+        void moveTo(ref WrappedBString other)
+        {
+            SysFreeString(other.ptr);
+            other.ptr = ptr;
+            ptr = null;
+        }
+    }
+
+    WrappedBString versionString, installDir;
+
     while (instances.Next(1, &instance, &fetched) == S_OK && fetched)
     {
-        BSTR bstrInstallDir;
-        if (instance.GetInstallationPath(&bstrInstallDir) != S_OK)
+        WrappedBString thisVersionString, thisInstallDir;
+        if (instance.GetInstallationVersion(&thisVersionString.ptr) != S_OK ||
+            instance.GetInstallationPath(&thisInstallDir.ptr) != S_OK)
             continue;
 
-        char[260] path;
-        int len = WideCharToMultiByte(CP_UTF8, 0, bstrInstallDir, -1, path.ptr, 260, null, null);
-        SysFreeString(bstrInstallDir);
-
-        if (len > 0)
-            return path[0..len].idup.ptr;
+        // FIXME: proper version strings comparison
```

Comment:
I'd add a comment explaining why it's ok for now, e.g. `existing versions are 15.0 to 16.5, problems expected in very distant future`

Reply:
We have the same problem for the comparison of the latest WinSDK subdir, but okay.